steward/src/cellars/cellar_v2_5.rs

```diff
@@ -0,0 +1,355 @@
+        }
+        Function::AddAdaptorToCatalogue(params) => {
+            validate_add_adaptor_to_catalogue(&cellar_id, &params.adaptor)?;
+            log_cellar_call(
+                CELLAR_NAME,
+                &AddAdaptorToCatalogueCall::function_name(),
+                &cellar_id,
+            );
+            let call = AddAdaptorToCatalogueCall {
+                adaptor: sp_call_parse_address(params.adaptor)?,
+            };
+
+            Ok(CellarV2_5Calls::AddAdaptorToCatalogue(call).encode())
+        }
+        Function::AddPositionToCatalogue(params) => {
+            validate_add_position_to_catalogue(&cellar_id, params.position_id)?;
+            log_cellar_call(
+                CELLAR_NAME,
+                &AddPositionToCatalogueCall::function_name(),
+                &cellar_id,
+            );
+            let call = AddPositionToCatalogueCall {
+                position_id: params.position_id,
+            };
+
+            Ok(CellarV2_5Calls::AddPositionToCatalogue(call).encode())
+        }
+        Function::RemoveAdaptorFromCatalogue(params) => {
+            log_cellar_call(
+                CELLAR_NAME,
+                &RemoveAdaptorFromCatalogueCall::function_name(),
+                &cellar_id,
+            );
+            let call = RemoveAdaptorFromCatalogueCall {
+                adaptor: sp_call_parse_address(params.adaptor)?,
+            };
+
+            Ok(CellarV2_5Calls::RemoveAdaptorFromCatalogue(call).encode())
+        }
+        Function::RemovePositionFromCatalogue(params) => {
+            log_cellar_call(
+                CELLAR_NAME,
+                &RemovePositionFromCatalogueCall::function_name(),
+                &cellar_id,
+            );
+            let call = RemovePositionFromCatalogueCall {
+                position_id: params.position_id,
+            };
+
+            Ok(CellarV2_5Calls::RemovePositionFromCatalogue(call).encode())
+        }
+        Function::IncreaseShareSupplyCap(params) => {
+            log_cellar_call(
+                CELLAR_NAME,
+                &IncreaseShareSupplyCapCall::function_name(),
+                &cellar_id,
+            );
+            let call = IncreaseShareSupplyCapCall {
+                new_share_supply_cap: string_to_u256(params.new_cap)?,
+            };
+
+            Ok(CellarV2_5Calls::IncreaseShareSupplyCap(call).encode())
+        }
+        Function::DecreaseShareSupplyCap(params) => {
+            log_cellar_call(
+                CELLAR_NAME,
+                &DecreaseShareSupplyCapCall::function_name(),
+                &cellar_id,
+            );
+            let call = DecreaseShareSupplyCapCall {
+                new_share_supply_cap: string_to_u256(params.new_cap)?,
+            };
+
+            Ok(CellarV2_5Calls::DecreaseShareSupplyCap(call).encode())
+        }
+    }
+}
+
+/// Encodes calls to the Adaptor contracts
+fn get_encoded_adaptor_calls(data: Vec<AdaptorCall>) -> Result<Vec<AbiAdaptorCall>, Error> {
```

Comment:
Is there any difference between this for 2.2 and 2.5? Could we in theory just use a common function for all v2 cellar types when encoding an adaptor call?

Reply:
each `AbiAdaptorCall` (return type) is uniquely scoped.